# Confidence interval

1. Help to estimate what actual value of the unknown population mean is
2. It is likely range for the true score of your entire population

# Margin of error

1. It indicates the range around the point estimate within which the true population parameter is likely to lie.
2. Margin of Error for a Population Mean (Large Sample Size):
Margin of Error = z * (population_std_dev / sqrt(sample_size))
3. Margin of Error for a Population Proportion:
Margin of Error = z * sqrt((sample_proportion * (1 - sample_proportion)) / sample_size)
4. large sample size - less error
5. It is halp of CI

### CI formula
1. Confidence Interval for a Population Mean (Large Sample Size, Known Population Standard Deviation):
1. Formula: Confidence Interval = sample_mean ± (z * (population_std_dev / sqrt(sample_size)))

2. Confidence Interval for a Population Mean (Large Sample Size, Unknown Population Standard Deviation):
2. Formula: Confidence Interval = sample_mean ± (z * (sample_std_dev / sqrt(sample_size)))

3. Confidence Interval for a Population Proportion:
3. Formula: Confidence Interval =sample_proportion ± (z * sqrt((sample_proportion * (1 - sample_proportion)) / sample_size))

In [1]:
import scipy.stats as stats

population_mean = 60
population_std = 8
sample_size = 50
sample_mean = 65
confidence_level = 0.9

standard_error = population_std / (sample_size ** 0.5)

critical_value = stats.t.isf((1 - confidence_level) / 2, df=sample_size - 1)

# Calculate the margin of error
margin_of_error = critical_value * standard_error

# Calculate the lower and upper bounds of the confidence interval
lower_bound = sample_mean - margin_of_error
upper_bound = sample_mean + margin_of_error

# Print the confidence interval
print("90% Confidence Interval: [{:.4f}, {:.4f}]".format(lower_bound, upper_bound))


90% Confidence Interval: [63.1032, 66.8968]


# Bayes Theorem

It allows us to update our beliefs or probabilities based on new evidence.
The theorem can be stated as follows:

P(A|B) = (P(B|A) * P(A)) / P(B)

Where:

1. P(A|B) is the posterior probability of event A given event B has occurred.
2. P(B|A) is the likelihood of event B given that event A has occurred.
3. P(A) is the prior probability of event A.
4. P(B) is the prior probability of event B.


### example 
Suppose there are two machines on a factory floor, Machine A and Machine B. Machine A produces 80% of the total output, while Machine B produces the remaining 20%. It is known that Machine A has a defect rate of 2%, while Machine B has a defect rate of 5%.

Now, let's say an item is randomly selected from the factory floor, and it is found to be defective. We want to calculate the probability that the item came from Machine A given this information.

In [2]:
# Prior probabilities
p_machine_a = 0.8  # Probability of selecting Machine A
p_machine_b = 0.2  # Probability of selecting Machine B

# Likelihoods
p_defect_given_a = 0.02  # Probability of defect given Machine A
p_defect_given_b = 0.05  # Probability of defect given Machine B

# Calculate the denominator (evidence) using the law of total probability
p_defect = (p_machine_a * p_defect_given_a) + (p_machine_b * p_defect_given_b)

# Calculate the posterior probability using Bayes' theorem
p_a_given_defect = (p_machine_a * p_defect_given_a) / p_defect

print("Probability that the item came from Machine A given it is defective:", p_a_given_defect)
print("Probability that the item came from Machine B given it is defective:", 1 - p_a_given_defect)


Probability that the item came from Machine A given it is defective: 0.6153846153846153
Probability that the item came from Machine B given it is defective: 0.3846153846153847


The result shows that there is approximately a 61.54% probability that the defective item came from Machine A, and a 38.46% probability that it came from Machine B.

# Type I error and Type II error

1. Type I Error: A Type I error, also known as a false positive, occurs when the null hypothesis is rejected even though it is true
2. example , This means concluding that the drug is effective when it has no actual effect.
3. Type II Error: A Type II error, also known as a false negative, occurs when the null hypothesis is not rejected even though it is false. 
4. example, This means failing to recognize the effectiveness of the drug and missing out on the potential benefits it could provide to patients. 
5. By setting a lower significance level (α), the probability of a Type I error decreases, but it increases the probability of a Type II error. Conversely, by setting a higher significance level, the probability of a Type II error decreases, but it increases the probability of a Type I error.